src/main/java/com/websafe/ccmsbe/service/PrivacyRegulationService.java

```diff
@@ -0,0 +1,84 @@
+package com.websafe.ccmsbe.service;
+
+import com.websafe.ccmsbe.entity.PrivacyRegulation;
+import com.websafe.ccmsbe.entity.Website;
+import com.websafe.ccmsbe.repository.PrivacyRegulationRepository;
+import com.websafe.ccmsbe.repository.WebsiteRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.util.List;
+import java.util.Optional;
+
+@Service
+public class PrivacyRegulationService {
+
+    private final PrivacyRegulationRepository privacyRegulationRepository;
+
+    private final WebsiteRepository websiteRepository;
+
+    @Autowired
+    public PrivacyRegulationService(PrivacyRegulationRepository privacyRegulationRepository, WebsiteRepository websiteRepository) {
+        this.privacyRegulationRepository = privacyRegulationRepository;
+        this.websiteRepository = websiteRepository;
+    }
+
+
+    public PrivacyRegulation createPrivacyRegulation(PrivacyRegulation privacyRegulation) {
+
+        return privacyRegulationRepository.save(privacyRegulation);
+    }
+
+    public List<PrivacyRegulation> getAllPrivacyRegulations() {
+        return privacyRegulationRepository.findAll();
+    }
+
+    public Optional<PrivacyRegulation> getPrivacyRegulationsFromWebsite(Long websiteId) {
+        return privacyRegulationRepository.findById(websiteId);
+    }
+
+    @Transactional
+    public String addPrivacyRegulationToWebsite(Long websiteId, Long regulationId) {
+        Website website = websiteRepository.findById(websiteId).orElse(null);
+        PrivacyRegulation privacyRegulation = privacyRegulationRepository.findById(regulationId).orElse(null);
+
+        //check for duplications
+
+        if (website == null || privacyRegulation == null) {
+            return "Unsuccessfully";
```

Comment:
Consider using a more structured error handling mechanism, which could involve throwing exceptions and handling them appropriately.